x/data/spec/06_tests.md

```diff
@@ -0,0 +1,79 @@
+
+- GIVEN - the content hash is empty
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash does not conform to the canonical IRI scheme, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash does not conform to the canonical IRI scheme
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of graph type and the canonicalization algorithm is unspecified, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of graph type and the canonicalization algorithm is unspecified
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of graph type and the canonicalization algorithm is not supported, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of graph type and the canonicalization algorithm is not supported
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of graph type and the canonicalization algorithm is not supported, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of graph type and the canonicalization algorithm is not supported
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of graph type and the merkle tree is not supported, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of graph type and the merkle tree is not supported
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of graph type and the digest length is not 256 bits, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of graph type and the digest length is not 256 bits
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of raw type and the media type is not supported, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of raw type and the media type is not supported
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+If a user tries to anchor data and the content hash is of raw type and the digest length is not 256 bits, then the transaction fails and the data is not anchored.
+
+- GIVEN - the content hash is of raw type and the digest length is not 256 bits
+- WHEN - user tries to anchor data
+- THEN - transaction fails, data is not anchored
+
+### Sign Data
```

Comment:
should we also include a test where a hash is passed but is not found on the chain?

Reply:
added in https://github.com/regen-network/regen-ledger/pull/673/commits/2998bb1e72c25eb9206933924b5e4f35354ba53c